### PR TITLE
Added trials passages and objectives json

### PR DESCRIPTION
**Pull request by asinusi**
This json will help with identifying what quests are for Trials and their respective objectives. Ideally I would only want one "Flawless" value in the json so I can flip the key/values and do something like `trialsJson.objectives["Rounds Won"] === objective.hash`.

### CODE REVIEW

**Comment by delphiactual** [comment]
filter out items with the `Dummy` ItemCategoryHash

**Comment by asinusi** [comment]
Filtering the items fixes the duplicate passages, but each of those items contain 4 `objectiveHashes`, 2 of them being the "Flawless" one.

**Comment by delphiactual** [comment]
`red pips` vs `green pips`

File: src/generate-trials-objectives.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    inventoryItem.displayProperties.name.startsWith('Passage of')
+  ) {
+    //Save the quest hash
+    trialsPassages.add(inventoryItem.hash);
+    //Now pull out each objective if we have any
+    inventoryItem.objectives?.objectiveHashes.forEach((o) => {
+      const obj = get('DestinyObjectiveDefinition', o);
+      if (obj) {
+        trialsObjectives[obj?.hash] = obj.displayProperties?.name || obj.progressDescription;
+      }
```
Comment (delphiactual) [inline src/generate-trials-objectives.ts:23]:
```suggestion
      if (obj) {
        if (obj.progressDescription === 'Flawless') {
          if (obj.completedValueStyle === 10) {
            trialsObjectives[obj?.hash] = obj.displayProperties?.name || obj.progressDescription;
          }
        } else {
          trialsObjectives[obj?.hash] = obj.displayProperties?.name || obj.progressDescription;
        }
      }
```

Comment (asinusi) [reply src/generate-trials-objectives.ts]: I understand now thanks for showing me.

**Pull request merged**